core/src/OpenACC/Kokkos_OpenACCSpace.hpp

```diff
@@ -85,16 +85,28 @@ class OpenACCSpace {
 template <>
 struct Kokkos::Impl::MemorySpaceAccess<Kokkos::HostSpace,
                                        Kokkos::Experimental::OpenACCSpace> {
+#if defined(KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE) && \
+    !defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU) && !defined(KOKKOS_ARCH_AMD_GPU)
```

Comment:
Let's only guard with `#ifdef KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE` and make that incompatible with a NVIDIA or AMD gpu architecture specified.
(same comment below)